sos/report/plugins/openshift.py

```diff
@@ -7,151 +7,369 @@
 # See the LICENSE file in the source distribution for further information.
 
 from sos.report.plugins import Plugin, RedHatPlugin
-import os.path
+from fnmatch import translate
+import os
+import re
 
 
-# This plugin handles OpenShift Enterprise versions up to 2.x
-# which are based in the upstream code of OpenShift Origin M4
-# located here: https://github.com/openshift/origin-server
-#
-# For later of OpenShift Origin based on: https://github.com/openshift/origin
-# like OpenShift Enterprise 3.x see the origin.py plugin
 class Openshift(Plugin, RedHatPlugin):
+    """This is the plugin for OCP 4.x collections. While this product is still
+    built ontop of kubernetes, there is enough difference in the collection
+    requirements and approach to warrant a separate plugin as opposed to
+    further extending the kubernetes plugin (or the OCP 3.x extensions included
+    in the Red Hat version of the kube plugin).
+
+    By default, this plugin will collect cluster information and inspect the
+    default namespaces/projects that are created during deployment - i.e. the
+    namespaces of the cluster projects matching openshift.* and kube.*. At the
+    time of this plugin's creation that number of default projects is already
+    north of 50; hence this plugin is expected to take a long time in both the
+    setup() and collect() phases. End-user projects may also be collected from
+    when those projects are included in the `add-namespaces` or
+    `only-namespaces` options.
+
+    It is expected to need to perform an `oc login` command in order for this
+    plugin to be able to correctly capture information, as system root is not
+    considered cluster root on the cluster nodes in order to access the API.
+
+    Users will need to either:
+
+        1) Provide the bearer token via the `-k openshift.token` option
+        2) Provide the bearer token via the `SOSOCPTOKEN` environment variable
+        3) Otherwise ensure that the root user can successfully run `oc` and
+           get proper output prior to running this plugin
+
 
-    short_desc = 'Openshift 2.x node and broker'
+    It is highly suggested that option #2 be used first, as this will prevent
+    the token from being recorded in output saved to the archive. Option #1 may
+    be used if this is considered an acceptable risk. It is not recommended to
+    rely on option #3, though it will provide the functionality needed.
+    """
+
+    short_desc = 'Openshift Container Platform 4.x'
 
     plugin_name = "openshift"
-    profiles = ('virt', 'openshift')
+    plugin_timeout = 600
```

Comment:
If you stated

> Total runtime of just the openshift plugin from start to finish averages around 10 minutes,

isn't the default timeout to 10 minutes too low?

Reply:
Our `plugin_timeout` doesn't include `setup()` phase time, so that's 10 minutes of `collect()` running. In my tests the `collect()` phase took 5-6 minutes, so we have some headroom there, but I have no problems increasing it further.